src/log.c

```diff
@@ -118,25 +118,52 @@ ly_errapptag(const struct ly_ctx *ctx)
     return NULL;
 }
 
-API struct ly_err_item *
-ly_err_new(LY_LOG_LEVEL level, LY_ERR no, LY_VECODE vecode, char *msg, char *path, char *apptag)
+API LY_ERR
+ly_err_new(struct ly_err_item **err, LY_ERR ecode, LY_VECODE vecode, char *path, char *apptag, const char *err_msg, ...)
 {
-    struct ly_err_item *eitem;
+    char *msg = NULL;
+    struct ly_err_item *e;
+
+    if (!err || ecode == LY_SUCCESS) {
+        /* nothing to do */
+        return ecode;
+    }
 
-    eitem = malloc(sizeof *eitem);
-    LY_CHECK_ERR_RET(!eitem, LOGMEM(NULL), NULL);
-    eitem->prev = eitem;
-    eitem->next = NULL;
+    va_list(print_args);
```

Comment:
It took me a few seconds to understand, why is it even possible to write a variable declaration this way? Please write it normally and put at the function start.

Reply:
it wasn't intended, it was actually just copy-pasted from the previous form of the function, fixed (not moved to the function start, but just into the block which needs it)